### PR TITLE
Tidy up some uses of `csearch::get_item_path`

### PR DESCRIPTION
**Pull request by mitaa**
(this incidentally fixes an error message where the paths separator is " " instead of "::")

### CODE REVIEW

File: src/librustc/metadata/csearch.rs [hunk 1]
```diff
@@ -95,6 +95,12 @@ pub fn get_item_path(tcx: &ty::ctxt, def: ast::DefId) -> Vec<ast_map::PathElem>
     })
 }
 
+pub fn get_item_name(tcx: &ty::ctxt, def: ast::DefId) -> ast::Name {
+    let cstore = &tcx.sess.cstore;
+    let cdata = cstore.get_crate_data(def.krate);
+    decoder::get_item_name(&*cstore.intr, &*cdata, def)
```
Comment (nrc) [inline src/librustc/metadata/csearch.rs]: do you need the `*`s here?

File: src/librustc/middle/ty.rs [hunk 2]
```diff
@@ -5704,13 +5700,10 @@ impl<'tcx> ctxt<'tcx> {
...
-                        let elems = csearch::get_item_path(self, found.def_id)
-                                        .into_iter()
-                                        .map(|p| p.to_string())
-                                        .collect::<Vec<_>>();
-
+                        let elems = csearch::get_item_path(self, found.def_id).into_iter();
                         self.sess.note(
-                            &format!("a second default is defined on `{}`", elems.join(" ")));
+                            &format!("a second default is defined on `{}`",
+                                path_to_string(elems)));
```
Comment (nrc) [inline src/librustc/middle/ty.rs]: nit: align `p` with `"`

**Comment by nrc** [comment]
lgtm, r+ with the nits addressed

**Comment by mitaa** [comment]
Nits addressed.

**Comment by arielb1** [comment]
@mitaa 

I would prefer to place `get_item_name` in `middle::ty` rather than csearch (including the same-crate case) - csearch isn't supposed to be accessed by random code.

**Comment by mitaa** [comment]
@arielb1 
Ahh, I see. (updated) 

I think this should be fine now.

**Comment by mitaa** [comment]
`csearch::get_item_name` still exists, but it is only called from the new method `item_name` on `ctxt`. 
I've kept it because it currently seems to be a convention that the `decoder` module is only accessed through other metadata modules like `csearch`.

**Comment by arielb1** [comment]
Could you trim the `is_local` checks around the places you changed?

**Comment by mitaa** [comment]
Updated. 

r? @arielb1

File: src/librustc/middle/intrinsicck.rs [hunk 3]
```diff
@@ -57,21 +55,7 @@ impl<'a, 'tcx> IntrinsicCheckingVisitor<'a, 'tcx> {
...
-            }
-        } else {
-            match csearch::get_item_path(self.tcx, def_id).last() {
-                Some(ref last) if intrinsic => {
-                    last.name() == "transmute"
-                }
-                _ => false,
-            }
-        }
+        intrinsic && self.tcx.item_name(def_id) == "transmute"
```
Comment (arielb1) [inline src/librustc/middle/intrinsicck.rs]: I guess we should check that this is indeed a foreign fn, but you can't really define `extern "rust-intrinsic" fn` statics anyway.

**Pull request merged**